bdk_rpc/src/lib.rs

```diff
@@ -0,0 +1,377 @@
+            client.create_wallet(&wallet_name, Some(true), None, None, None)?;
+        } else {
+            // TODO: move back to api call when https://github.com/rust-bitcoin/rust-bitcoincore-rpc/issues/225 is closed
+            let args = [
+                Value::String(wallet_name.clone()),
+                Value::Bool(true),  // disable_private_keys
+                Value::Bool(false), //blank
+                Value::Null,        // passphrase
+                Value::Bool(false), // avoid reuse
+                Value::Bool(false), // descriptor
+            ];
+            let _: Value = client.call("createwallet", &args)?;
+        }
+
+        info!("wallet created: {}", wallet_name);
+    }
+
+    let blockchain_info = client.get_blockchain_info()?;
+    let network = match blockchain_info.chain.as_str() {
+        "main" => Network::Bitcoin,
+        "test" => Network::Testnet,
+        "regtest" => Network::Regtest,
+        "signet" => Network::Signet,
+        _ => {
+            return Err(RpcError::Generic(
+                "Invalid network in blockchain info".to_string(),
+            ))
+        }
+    };
+    if network != config.network {
+        return Err(RpcError::InvalidNetwork {
+            requested: config.network,
+            found: network,
+        });
+    }
+
+    Ok(client)
+}
+
+/// Sync the Tracker, Sparsechain and Client with each other.
+pub fn rpc_sync<K: Debug + Clone + Ord>(
+    client: &Client,
+    tracker: &mut KeychainTracker<K>,
+    sparse_chain: &mut SparseChain,
+) -> Result<(), RpcError> {
+    let start = std::time::Instant::now();
+
+    // Import spks from Tracker to Client
+    // Can have spks that are already imported
+    let spk_map = &*tracker.script_pubkeys();
+    let _labelled_scripts: Vec<(Option<String>, Script)> = spk_map
+        .iter()
+        .map(|((keychain, _), script)| {
+            let label = Some(format!("{:?}", keychain));
+            (label, script.clone())
+        })
+        .collect();
+
+    let labelled_scripts = _labelled_scripts
+        .iter()
+        .map(|(label, script)| {
+            let label = label.as_ref().map(|l| l.as_str());
+            (label, script)
+        })
+        .collect();
+    import_multi(&client, labelled_scripts)?;
+
+    // Import spks from Client back to Tracker
+    // This is useful when a fresh Tracker is started but it previously
+    // imported spks into Client.
+    let mut keychain_derive_map = Vec::new();
+    for (keychain, _) in &mut tracker.iter_keychains(..) {
+        let label = Value::String(format!("{:?}", keychain));
+        if let Ok(addresses) = client.call::<Value>("getaddressesbylabel", &[label]) {
+            let max_index = addresses
+                .as_object()
+                .expect("Address object expected")
+                .iter()
+                .count()
+                - 1;
+            keychain_derive_map.push((keychain, max_index));
+        }
+    }
```

Comment:
I think syncing derivation indexes from RPC to client probably belongs in a different function. It would be great if we only passed `sparse_chain` into this function and a list of labels you want to sync against.